### ISSUE TITLE
docs: QEMU Command Line Documentation

### ISSUE THREAD

**Issue by jpecholt** [kata-containers/kata-containers#4948]
__Issue:__
I was implementing support for SEV-SNP VMs and struggled with the QEMU command line. I understand that the configuration is complex and certain components / options are only added when specific hardware is used, the system's software supports it and respective configurations are made. Other components are absolutely necessary and cannot be omitted. While implementing SEV-SNP support, I was struggling to (experimentally) decrease the command line to the minimal amount of components to ensure that no other components are possibly causing bugs due to lacking support or the like. 

__Proposal:__
Personally, I would have greatly benefited from a Readme of sorts describing which device and which option relates to what component in the Kata-Containers world, which components are optional and can be removed by changing certain values in the configuration file (e.g., shared memory), which are always added but not critical (e.g., logging) and which are absolutely necessary to the deployment of a container (e.g., socket to communicate with kata-agent).
One option would be to collect an entire QEMU command, separate it into its options and shortly explain what each of them is for, where it can be reconfigured and if it is optional or necessary

**Comment by c3d** [comment]
@jpecholt This is a very good idea. I think that you are talking specifically about the options as sent by the Kata runtime, not about the general `qemu` options, right?

**Comment by jpecholt** [comment]
Yes, indeed. The QEMU options themselves are well-documented in my eyes. This is more about the options with respect to Kata-Containers / the Kata runtime: What Kata feature does an option facilitate, which option (e.g., a device) can be substituted by another or left out entirely and where can this configuration be made (e.g., configuration file value `shared_fs`), which option is crucial to the running of Kata-Containers...
